services/user.py

```diff
@@ -0,0 +1,59 @@
+from django.db import transaction
+
+from db.models import User
+
+
+def create_user(
+    username: str,
+    password: str,
+    email: str = None,
+    first_name: str = None,
+    last_name: str = None,
```

Comment:
When you have default values, indicate that your arguments are optional in type annotation. Apply this change throughout your code
```suggestion
    email: str | None = None,
    first_name: str | None = None,
    last_name: str | None = None,
```